lib/elixir/lib/calendar.ex

```diff
@@ -0,0 +1,87 @@
+defmodule Calendar do
+  @moduledoc """
+  This module defines the responsibilities for working with
+  calendars, dates, times and datetimes in Elixir.
+
+  Currently it defines types but may define the calendar
+  behaviour in future versions. For the actual date, time
+  and datetime structures, see `Date`, `Time`, `NaiveDateTime`
+  and `DateTime`.
+
+  Note the year, month, day, etc designations are over specified
+  (i.e. an integer instead of 1..12 for months) because different
+  calendars may have a different number of numbers and so on.
+  """
+
+  @type year        :: integer
+  @type month       :: integer
+  @type day         :: integer
+  @type hour        :: integer
+  @type minute      :: integer
+  @type second      :: integer
+  @type microsecond :: integer
```

Comment:
Types for hour, minute, second, and microsecond could all be made `non_neg_integer`.